### DISCUSSION TITLE
Apps only installed not activated?

### DISCUSSION THREAD

**Discussion by loeffelpan** [nextcloud/all-in-one#1649] [Ideas] [answered]
Finally I got AIO running.
I added an environment variable to install apps that are used in my instance. But they were only installed not activated.
Not a big deal to do this manually, but I would suggest that apps that I explicitly add to my docker run command / compose file will be installed **and activated**.

`NEXTCLOUD_STARTUP_APPS=calendar contacts photos files_external twofactor_totp`

```
calendar 4.2.0 installed
calendar enabled
contacts 5.0.2 installed
contacts enabled
photos already installed
files_external already installed
twofactor_totp already installed
```

**Comment by szaimen** [comment]
Hi, this is due to these apps being delivered with Nextcloud out-of-the-box. So for these we could add an occ app:enable.

**Comment by szaimen** [comment, marked as answer]
This will be fixed with https://github.com/nextcloud/all-in-one/pull/1654

**Comment by loeffelpan** [comment]
Wonderful.
Maybe as an addition it would ne helpful to declare app, which should be uninstalled/deactivated.
E. g. `NEXTCLOUD_STARTUP_APPS=contacts calendar -circles -dashboard`.

**Reply by szaimen** [reply]
also implemented with the PR above. I'll let you know when the feature is available in the beta channel

**Reply by szaimen** [reply]
This is now released with v4.1.0 Beta. Testing and feedback is welcome! See https://github.com/nextcloud/all-in-one#how-to-switch-the-channel

**Reply by loeffelpan** [reply]
I tried to test this. But don't know how. Stopping an prune the containers and recreate them doesn't lead to app enabling/disabling from environment variable. Do I have to reset the whole instance and create a new one from backup?

**Reply by szaimen** [reply]
This only works on the first installation of AIO. So only with a clean instance.

**Reply by loeffelpan** [reply]
Restore from backup won't use that variable either? Only new instances?

**Reply by szaimen** [reply]
Yes, only new instances.

**Comment by loeffelpan** [comment]
Testing done. Working fine. Thanks for implementation.
But there is one thing maybe interesting for your documentation.
The string for `NEXTCLOUD_STARTUP_APPS` can get long in some cases and maybe someone would use multiline strings for that.
This should word in docker run-commands and in compose files with an array as `environment:`(see below). But be carefult with a `environment:` dictonary (see below)
In that case it is important to use `>` or `>-` before the multiline statement. If using `|` there will be newlines `\n` and app-list will only be respected until the first `\n`.

Array:
```
    environment:
      [...]
      - NEXTCLOUD_STARTUP_APPS=
          calendar contacts tasks photos files_external twofactor_totp 
          -activity -admin_audit -circles -contactsinteraction -comments 
          -dashboard -federation -files_versions -firstrunwizard -nextcloud_announcements
          -password_policy -privacy -recommendations -related_ressources -sharebymail
          -survey_client -systemtags -text -user_status -weather_status
```

Dictonary:
```
    environment:
      [...]
      NEXTCLOUD_STARTUP_APPS: >-
          calendar contacts tasks photos files_external twofactor_totp 
          -activity -admin_audit -circles -contactsinteraction -comments 
          -dashboard -federation -files_versions -firstrunwizard -nextcloud_announcements
          -password_policy -privacy -recommendations -related_ressources -sharebymail
          -survey_client -systemtags -text -user_status -weather_status
```
